### DISCUSSION TITLE
[Feature request w/ contribution] Provide a way to access the Firebase instance id

### DISCUSSION THREAD

**Discussion by ghost** [firebase/flutterfire#5600] [Feature request]
Hi 👋

I need a way to access the [Firebase instance id](https://firebase.google.com/docs/reference/android/com/google/firebase/iid/FirebaseInstanceId#getId()) in Flutter, and am willing to contribute the plugin implementation for it. What is the best place to add this API? The [`firebase_messaging`](https://github.com/FirebaseExtended/flutterfire/tree/4d7b2bdd7fbdee65551754968b140653ea16128f/packages/firebase_messaging) plugin already has a `deleteInstanceID` method, so maybe `getInstanceID` belongs there? https://github.com/FirebaseExtended/flutterfire/blob/4d7b2bdd7fbdee65551754968b140653ea16128f/packages/firebase_messaging/lib/firebase_messaging.dart#L166-L173

**Comment by Nathan-Heller** [comment]
This would also be very useful for the `in_app_messaging` feature, since the InstanceID is required to test sending an in-app message via the console.
